Sources/Workspace/Workspace.swift

```diff
@@ -1062,14 +1066,22 @@ extension Workspace {
     }
 
     /// This enum represents state of an external package.
-    fileprivate enum PackageStateChange {
+    fileprivate enum PackageStateChange: Equatable {
         /// The requirement imposed by the the state.
-        enum Requirement {
+        enum Requirement: Equatable {
             /// A version requirement.
             case version(Version)
 
             /// A revision requirement.
             case revision(Revision, branch: String?)
+            
+            static func == (lhs: Requirement, rhs: Requirement) -> Bool {
+                switch (lhs, rhs) {
+                case (.version(let a), .version(let b)): return a == b
+                case (.revision(let a), .revision(let b)): return (a.0 == b.0) && (a.1 == b.1)
+                default: return false
```

Comment:
Instead of default, write an exhaustive switch statement. It will prevent missed sites if the enum expands in future.